### PR TITLE
Group admin cmds

### LINKED ISSUE #3424: Group admin cmds similar to oc cmds
I think it will be helpful for the admin if we can organize all supported cmds in oadm -h.
We can do something like,

```
Basic Commands:
  new-project                  Create a new project
  policy                            Manage authorization policy

Install Commands:
  router                            Install an OpenShift router
  registry                          Install the OpenShift Docker registry
  ipfailover                       Install an IP failover group to a set of nodes

Maintenance Commands:
  build-chain                       Output build dependencies of a specific image stream
  prune                               Remove older versions of resources from the server
  manage-node                  Manage nodes - list pods, evacuate, or mark ready

Settings Commands:
  config                                Change configuration files for the client
  create-api-client-config     Create a config file for connecting to the server as a user
  create-kubeconfig             Create a basic .kubeconfig file from client certs
  
Advanced Commands:  
  create-bootstrap-project-template Create bootstrap project template for OpenShift
  create-bootstrap-policy-file    Create bootstrap policy for OpenShift
  overwrite-policy                      Reset the policy to the default values
  create-node-config                Create a configuration bundle for a node
  create-master-certs               Create certificates for an OpenShift master  
  create-key-pair                      Create a public/private key pair
  create-server-cert                  Create a signed server certificate and key
  create-signer-cert                  Create a signer (certificate authority/CA) certificate
```
/cc @smarterclayton @danmcp

### PR DESCRIPTION
**Pull request by pravisankar**
- Group commands in oadm for ease of use
- Group oadm certificate related cmds under 'oadm ca'

closes #3424 , #2461

### CODE REVIEW

**Comment by pravisankar** [comment]
@liggitt @smarterclayton PTAL

**Comment by liggitt** [comment]
I would keep the more descriptive names (`create-master-certs` vs `master`). Can we leave the old commands functional but deprecated and unsurfaced in the list of command groups?

**Comment by pravisankar** [comment]
[test]

**Comment by pravisankar** [comment]
@liggitt updated, ptal

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -283,6 +283,9 @@ echo "templates: ok"
 [ "$(openshift kubectl 2>&1 | grep 'Kubernetes cluster')" ]
 [ "$(oadm 2>&1 | grep 'OpenShift Administrative Commands')" ]
 [ "$(openshift admin 2>&1 | grep 'OpenShift Administrative Commands')" ]
+[ "$(oadm | grep 'Basic Commands:')" ]
+[ "$(oadm | grep 'Install Commands:')" ]
+[ "$(oadm ca | grep 'Manage certificates')" ]
 [ "$(openshift start kubernetes 2>&1 | grep 'Kubernetes server components')" ]
```
Comment (liggitt) [inline hack/test-cmd.sh]: Add a group of test invocations to make sure the old commands still exist. Just invoke --help on them and grep for an expected help string for `openshift admin create-master-certs`, etc

File: pkg/cmd/templates/templater.go [hunk 2]
```diff
@@ -40,7 +40,8 @@ func (g CommandGroups) Has(c *cobra.Command) bool {
 func AddAdditionalCommands(g CommandGroups, message string, cmds []*cobra.Command) CommandGroups {
 	group := CommandGroup{Message: message}
 	for _, c := range cmds {
-		if !g.Has(c) {
+		// Don't show commands that has no short description
+		if !g.Has(c) && len(c.Short) != 0 {
```
Comment (liggitt) [inline pkg/cmd/templates/templater.go]: @fabianofranz @deads2k I'm not as familiar with this part... any downside to this?

File: hack/test-cmd.sh [hunk 3]
```diff
@@ -112,7 +112,7 @@ do
     SERVER_HOSTNAME_LIST="${SERVER_HOSTNAME_LIST},${IP_ADDRESS}"
 done <<< "${ALL_IP_ADDRESSES}"
 
-openshift admin create-master-certs \
+openshift admin ca create-master-certs \
```
Comment (deads2k) [inline hack/test-cmd.sh]: Be sure to add an equivalent line back using `openshift admin create-master-certs`.  If we have to be backwards compatible, we need to leave the old command in there to be sure we never break it.

File: pkg/cmd/templates/templater.go (hunk 2 above)
Comment (fabianofranz) [inline pkg/cmd/templates/templater.go]: Looks good.

**Comment by pravisankar** [comment]
@liggitt added deprecated cmd checks in hack/test-cmd.sh, please review/merge

**Comment by liggitt** [comment]
LGTM. Squash?

**Comment by pravisankar** [comment]
[merge]

**Comment by pravisankar** [comment]
re[merge]

**Pull request merged**